src/lib/y2network/config_reader/sysconfig_interfaces.rb

```diff
@@ -0,0 +1,154 @@
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2network/interface"
+require "y2network/virtual_interface"
+require "y2network/physical_interface"
+require "y2network/connection_config/ethernet"
+
+Yast.import "LanItems"
+Yast.import "NetworkInterfaces"
+
+module Y2Network
+  module ConfigReader
+    # This class reads interfaces configuration from sysconfig files
+    #
+    # * Physical interfaces are read from the hardware.
+    # * Virtual interfaces + Connections are read from sysconfig.
+    #
+    # @see Y2Network::Interface
+    # @see Y2Network::Connection::Connection
+    class SysconfigInterfaces
+      # Returns the interfaces and connections configuration
+      #
+      # @todo Should we use different readers?
+      # @todo Are virtual interfaces coming from connections only?
+      def config
+        [interfaces, connections]
+      end
+
+      # List of interfaces
+      #
+      # @return [InterfacesCollection]
+      def interfaces
+        @interfaces ||= Y2Network::InterfacesCollection.new(physical_interfaces) # + virtual_interfaces
+      end
+
+      # List of virtual interfaces
+      #
+      # @return [Array<Interface>]
+      def virtual_interfaces
+        @virtual_interfaces ||= connections.map(&:interface).compact
+      end
+
+      # @return [Array<ConnectionConfig>]
+      def connections
+        configured_devices.map do |name|
+          interface = physical_interfaces.find { |i| i.name == name }
+          conn_type = interface ? interface.type : find_type_for(name)
+          # TODO: it may happen that the interface does not exist yet (hotplug, usb, or whatever)
+          # How should we handle those cases?
+          connection_config_for(conn_type, name)
+        end
+      end
+
+    private
+
+      # Returns the physical interfaces
+      #
+      # Physical interfaces are read from the old LanItems module
+      #
+      # @return [Array<Interface>]
+      def physical_interfaces
+        @physical_interfaces ||= Yast::LanItems.Hardware.map do |card|
+          iface = Y2Network::PhysicalInterface.new(card["dev_name"])
+          iface.description = card["name"]
+          iface.type = card["type"].to_sym
+          iface
+        end
+      end
+
+      # @return [Regex] expression to filter out invalid ifcfg-* files
+      IGNORE_IFCFG_REGEX = /(\.bak|\.orig|\.rpmnew|\.rpmorig|-range|~|\.old|\.scpmbackup)$/
+
+      # List of devices which has a configuration file (ifcfg-*)
+      #
+      # @return [Array<String>] List of configured devices
+      def configured_devices
+        files = Yast::SCR.Dir(Yast::Path.new(".network.section"))
+        files.reject { |f| IGNORE_IFCFG_REGEX =~ f || f == "lo" }
```

Comment:
interesting, I do not know that regex also has `=~` operand. I always use `string =~ /regexp/`